dogs-initial/src/main/java/com/lambdaschool/dogsinitial/controller/DogController.java

```diff
@@ -8,35 +9,71 @@
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.ModelAndView;
 
 import java.util.ArrayList;
 
 @RestController
 @RequestMapping("/dogs")
-public class DogController
-{
+public class DogController {
     // localhost:8080/dogs/dogs
     @GetMapping(value = "/dogs")
-    public ResponseEntity<?> getAllDogs()
-    {
+    public ResponseEntity<?> getAllDogs() {
         return new ResponseEntity<>(DogsinitialApplication.ourDogList.dogList, HttpStatus.OK);
     }
 
     // localhost:8080/dogs/{id}
     @GetMapping(value = "/{id}")
-    public ResponseEntity<?> getDogDetail(@PathVariable long id)
-    {
+    public ResponseEntity<?> getDogDetail(
+            @PathVariable
+                    long id) {
+
         Dog
-                rtnDog = DogsinitialApplication.ourDogList.findDog(d -> (d.getId() == id));
+                rtnDog;
+        if (DogsinitialApplication.ourDogList.findDog(d -> (d.getId() ==
+                id)) ==
+                null) {
+            throw new ResourceNotFoundException("The dog with the id of " +
+                    id +
+                    " does not exist");
+        } else {
```

Comment:
Remember that you don't need to place an `else` block here since the `throw` statement stops the method's execution already (though it is still fine to include it if preferred for being declarative).